### DISCUSSION TITLE
v0.0.15 SOCKS5 client with username/password authentication

### DISCUSSION THREAD

**Discussion by fortuna** [OutlineFoundation/outline-sdk#205] [Announcements]
Our [SOCKS5 client library](https://pkg.go.dev/github.com/Jigsaw-Code/outline-sdk/transport/socks5) now supports basic [username/password authentication](https://datatracker.ietf.org/doc/html/rfc1929) via `socks5.StreamDialer.SetCredentials(username, password)`. 

This makes SOCKS5 suitable for public proxy usage when paired with secure transports like TLS. The Outline SDK provides the necessary libraries for both SOCKS5-over-TLS and SOCKS5-over-Shadowsocks transport.

Thanks @amircybersec for the contribution: https://github.com/Jigsaw-Code/outline-sdk/pull/189

**Full Changelog**: https://github.com/Jigsaw-Code/outline-sdk/compare/v0.0.14...v0.0.15

<hr /><em>This discussion was created from the release <a href='https://github.com/Jigsaw-Code/outline-sdk/releases/tag/v0.0.15'>v0.0.15 SOCKS5 client with username/password authentication</a>.</em>

**Comment by amircybersec** [comment]
This is exciting! I have setup a proxy server to use SOCKS5-over-TLS and the config goes something like this:

```split:2|tlsfrag:5|tls:sni=open.com|socks5://user:pass@secretdomain.com:443|split:2```

The packet splits and TLS fragmentation makes fingerprinting based on packet sizes more difficult.

I can tunnel my traffic through it. My server uses `socat` to perform SSL handshake and passed the traffic to a local `socks5` server (running on port 8000 in this example) after decrypting it. I used `certbot` to create a CA-signed certificate. 

```
socat -dd openssl-listen:443,reuseaddr,cert=fullchain.pem,key=privkey.pem,verify=0,fork tcp-connect:localhost:8080
```

I will publish a short tutorial on this very soon.

**Reply by fortuna** [reply]
I'm looking forward to that tutorial! :-)
